sdk/tables/Azure.Data.Tables/tests/TableClientLiveTests.cs

```diff
@@ -388,6 +389,104 @@ public async Task EntityDeleteRespectsEtag()
+        /// </summary>
+        [Test]
+        public async Task InsertedEntitiesAreRoundtrippedWithProperOdataAnnoations()
+        {
+            List<IDictionary<string, object>> entityResults;
+            List<Dictionary<string, object>> entitiesToInsert = CreateTableEntities(PartitionKeyValue, 1);
+
+            // Insert the new entities.
+
+            foreach (var entity in entitiesToInsert)
+            {
+                await client.InsertAsync(entity).ConfigureAwait(false);
+            }
+
+            // Query the entities with a filter specifying that to RowKey value must be greater than or equal to '10'.
+
+            entityResults = (await client.QueryAsync(filter: $"PartitionKey eq '{PartitionKeyValue}' and RowKey eq '01'").ToEnumerableAsync().ConfigureAwait(false)).ToList();
+
+            Assert.That(entityResults.First()[StringTypePropertyName], Is.TypeOf<string>(), "The entity property should be of type string");
+            Assert.That(entityResults.First()[DateTypePropertyName], Is.TypeOf<DateTime>(), "The entity property should be of type DateTime");
+            Assert.That(entityResults.First()[GuidTypePropertyName], Is.TypeOf<Guid>(), "The entity property should be of type Guid");
+            Assert.That(entityResults.First()[BinaryTypePropertyName], Is.TypeOf<byte[]>(), "The entity property should be of type byte[]");
+            Assert.That(entityResults.First()[Int64TypePropertyName], Is.TypeOf<long>(), "The entity property should be of type int64");
+            Assert.That(entityResults.First()[DoubleTypePropertyName], Is.TypeOf<double>(), "The entity property should be of type double");
+            Assert.That(entityResults.First()[DoubleDecimalTypePropertyName], Is.TypeOf<double>(), "The entity property should be of type double");
+            Assert.That(entityResults.First()[IntTypePropertyName], Is.TypeOf<int>(), "The entity property should be of type int");
+        }
+
+        /// <summary>
+        /// Validates the functionality of the TableClient.
+        /// </summary>
+        [Test]
+        public async Task UpsertedEntitiesAreRoundtrippedWithProperOdataAnnoations()
+        {
+            List<IDictionary<string, object>> entityResults;
+            List<Dictionary<string, object>> entitiesToInsert = CreateTableEntities(PartitionKeyValue, 1);
+
+            // Insert the new entities.
+
+            foreach (var entity in entitiesToInsert)
+            {
+                await client.UpsertAsync(entity).ConfigureAwait(false);
+            }
+
+            // Query the entities with a filter specifying that to RowKey value must be greater than or equal to '10'.
+
+            entityResults = (await client.QueryAsync(filter: $"PartitionKey eq '{PartitionKeyValue}' and RowKey eq '01'").ToEnumerableAsync().ConfigureAwait(false)).ToList();
+
+            Assert.That(entityResults.First()[StringTypePropertyName], Is.TypeOf<string>(), "The entity property should be of type string");
```

Comment:
Does round tripping fail without annotations?

Reply:
annotations are added on serialization and removed on de-serialization so that they are transparent to the developer.

Reply:
Right, but what happens if we don't add them?

Reply:
the worst thing that happens is that everything that is sent as a string type is interpreted as a string rather than guid, dateTime, etc.

Reply:
Some people might consider that data corruption.  It's still retrievable, but definitely not what the customer would expect to happen.

Reply:
Yes, this is what these tests aim to validate doesn’t ever happen.